search/node-search_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package search
+
+import (
+	"errors"
+	"strconv"
+	"testing"
+
+	"gopkg.in/yaml.v3"
+)
+
+func performSearch(t *testing.T, filePath string, seeking Seeking, yamlPaths ...[]string) []*yaml.Node {
+	var foundNodes []*yaml.Node
+	baseNode, parseNode := MakeNode(filePath)
+	if parseNode != nil {
+		t.Fatalf(parseNode.Error())
+	}
+	for _, yamlPath := range yamlPaths {
+		node, searchError := findComponent(baseNode.Content[0], seeking, yamlPath...)
+		if searchError != nil {
+			t.Errorf(searchError.Error())
+		} else {
+			foundNodes = append(foundNodes, node)
+		}
+	}
+	return foundNodes
+}
+
+func validateIndex(sequenceNode *yaml.Node, ind int) error {
+	if sequenceNode.Kind != yaml.SequenceNode {
+		println(sequenceNode.Value)
+		return errors.New("not a sequence node")
+	}
+	if ind < 0 || ind >= len(sequenceNode.Content) {
+		return errors.New("not in sequence bounds")
+	}
+	return nil
+}
+
+// returns a slice of the last string from each path, order is perserved
+func getLastTokensFromPaths(yamlPaths ...[]string) []string {
+	var lastStrings []string
+	for _, token := range yamlPaths {
+		lastStrings = append(lastStrings, token[len(token)-1])
+	}
+	return lastStrings
+}
+
+// Assert that known paths are reported in Search
+func TestHardcodedPaths(t *testing.T) {
```

Comment:
Can we think of a better name.  The term Hardcoded implies that we have specific values in business logic our code which are coded.  That doesn't seem to be the what we are testing here.

Reply:
renamed to testkeysearch